orchestra/tests/test_task_lifecycle.py

```diff
@@ -172,117 +177,246 @@ def test_worker_has_reviewer_status(self):
+
+    def test_revert_task_to_datetime(self):
+        test_start = timezone.now()
+
+        def _sample_complete_task():
+            task = Task.objects.create(
+                project=self.projects['empty_project'],
+                status=Task.Status.COMPLETE,
+                step=self.test_step,
+                start_datetime=test_start)
+
+            workers = {
+                'entry': self.workers[0],
+                'reviewer': self.workers[1]
+            }
+
+            assignments = [
+                {
+                    'worker': workers['entry'],
+                    'start_datetime': test_start + timedelta(hours=1)
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'start_datetime': test_start + timedelta(hours=3)
+                }
+            ]
+
+            snapshots = [
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=2),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.REJECT,
+                    'datetime': test_start + timedelta(hours=4),
+                },
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=5),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.ACCEPT,
+                    'datetime': test_start + timedelta(hours=6),
+                },
+            ]
+
+            for i, assignment in enumerate(assignments):
+                TaskAssignment.objects.create(
+                    worker=assignment['worker'], task=task,
+                    status=TaskAssignment.Status.SUBMITTED,
+                    in_progress_task_data={},
+                    assignment_counter=i,
+                    snapshots={'snapshots': [], '__version__': 2},
+                    start_datetime=assignment['start_datetime'])
+
+            for i, snapshot in enumerate(snapshots):
+                assignment = task.assignments.get(worker=snapshot['worker'])
+                assignment.snapshots['snapshots'].append({
+                    'datetime': snapshot['datetime'],
+                    'type': snapshot['type'],
+                    'work_time_seconds': 0,
+                    'data': {
+                        snapshot['worker'].user.username: snapshot['type']
+                    },
+                })
+                assignment.save()
+
+            return task
+
+        def _test_reverted_task(datetime, status, num_assignments,
+                                num_snapshots_per_assignment):
+            task = _sample_complete_task()
```

Comment:
I don't think there's a ton of benefit, since the initial status of the task doesn't factor in anywhere.